File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/LootToUI.java
/**
 * @file LootToUI.java
 * @author Profesor
 */
package deepspace;

/**
 * @brief Representación ToUI de Loot
 */
public class LootToUI {
    private final int nSupplies;
    private final int nWeapons;
    private final int nShields;
    private final int nHangars;
    private final int nMedals;

    /**
     * @brief Constructor con un parámetro
     * @param l Instancia de la clase Loot
     */
    LootToUI(Loot l) {
        nSupplies=l.getNSupplies();
        nWeapons=l.getNWeapons();
        nShields=l.getNShields();
        nHangars=l.getNHangars();
        nMedals=l.getNMedals();
    }  

    /**
     * @brief Consultor del número de suministros
     * @return nSupplies
     */
    public int getnSupplies() {
        return nSupplies;
    }

    /**
     * @brief Consultor del número de armas
     * @return nWeapons
     */
    public int getnWeapons() {
        return nWeapons;
    }

    /**
     * @brief Consultor del número de escudos
     * @return nShields
     */
    public int getnShields() {
        return nShields;
    }

    /**
     * @brief Consultor del número de hangares
     * @return nHangars
     */
    public int getnHangars() {
        return nHangars;
    }

    /**
     * @brief Consultor del número de medallas
     * @return nMedals
     */
    public int getnMedals() {
        return nMedals;
    }  
}

File: practicas/practica_2/Deepspace_Ruby_P2/lib/LootToUI.rb
#encoding:utf-8

# File: LootToUI.rb
# Author: Profesor

module Deepspace

# Brief: Representación ToUI de Loot
class LootToUI

  # Consultores
  attr_reader :nSupplies, :nWeapons, :nShields, :nHangars, :nMedals

  # Brief: Constructor con un parámetro
  # Param l: Instancia de la clase Loot
  def initialize(l)
    @nSupplies=l.nSupplies
    @nWeapons=l.nWeapons
    @nShields=l.nShields
    @nHangars=l.nHangars
    @nMedals=l.nMedals
  end

  # Brief: Función para representar una instancia de la clase LootToUI
  #        en un string
  # Return: out String que representa una instancia de la clase LootToUI
  def to_s
    return "Supplies: #{@nSupplies}, Weapons: #{@nWeapons}, Shields: #{@nShields}, Hangars: #{@nHangars}, Medals: #{@nMedals}"
  end

end # Class

end # Module